platform/view/services/comm/builder.go

```diff
@@ -33,16 +33,18 @@ func newHost(ListenAddress string, keyDispenser PrivateKeyDispenser) (*P2PNode,
 		return nil, err
 	}
 
-	ctx := context.Background()
-
-	host, err := libp2p.New(ctx,
+	opts := []libp2p.Option{
 		libp2p.ListenAddrs(addr),
 		libp2p.Identity(priv),
-	)
+		libp2p.ForceReachabilityPublic(),
+	}
+
+	host, err := libp2p.New(opts...)
 	if err != nil {
 		return nil, err
 	}
 
+	ctx := context.Background()
```

Comment:
Shall we pass the context to give the caller of the function to have a chance to interrupt the process? If the passed context is nil, then context.Background() is used.

Reply:
good idea - let's cover this in a separate PR. There are a few improvements we could add to the comm services.

Reply:
Okay!